power-sequencer/main.cpp

```diff
@@ -51,21 +54,30 @@ int main(int argc, char** argv)
     auto bus = sdbusplus::bus::new_default();
     bus.attach_event(event.get(), SD_EVENT_PRIORITY_NORMAL);
 
-    auto device = std::make_unique<UCD90160>(0, bus);
-
     std::unique_ptr<DeviceMonitor> monitor;
 
     if (action == "pgood-monitor")
     {
         // If PGOOD doesn't turn on within a certain
         // time, analyze the device for errors
+        auto device = std::make_unique<UCD90160>(0, bus);
+        monitor = std::make_unique<PGOODMonitor>(std::move(device), bus, event,
+                                                 interval);
+    }
+    else if (action == "mihawk-cpld-pgood-monitor")
+    {
+        // Add new action for Mihawk:
+        // If PGOOD doesn't turn on within a certain
+        // time, analyze CPLD for errors
+        auto device = std::make_unique<MihawkCPLD>(0, bus);
         monitor = std::make_unique<PGOODMonitor>(std::move(device), bus, event,
                                                  interval);
     }
     else // runtime-monitor
     {
         // Continuously monitor this device both by polling
         // and on 'power lost' signals.
+        auto device = std::make_unique<UCD90160>(0, bus);
```

Comment:
Maybe you need to add runtime-monitor for mihawk

Reply:
Mihawk's CPLD function doesn't get as much "runtime" information as UCD. 
I think this feature can be discussed again. 
It is expected to be added later version.